metaflow/plugins/kfp/kfp.py

```diff
@@ -227,21 +227,71 @@ def create_kfp_pipeline_yaml(self, pipeline_file_path) -> str:
             pipeline_conf=pipeline_conf,
         )
 
+        self.update_kfp_yaml(  # Make flow_parameters_json optional. This aligns with Metaflow CLI's behavior.
+            pipeline_file_path,
+            [
+                (
+                    "spec.arguments.parameters",
+                    [{"name": "flow_parameters_json", "value": "{}"}],
+                ),
+            ],
+        )
+
         # re-write the workflow serviceAccountName if metaflow KUBERNETES_SERVICE_ACCOUNT
         # is set.
         if KUBERNETES_SERVICE_ACCOUNT:
-            self.set_kfp_client()
-            # use kfp client extract yaml method so we do not recreate logic that accounts
-            # for various extensions supported by kfp
-            workflow_yaml = self._client._extract_pipeline_yaml(pipeline_file_path)
-            workflow_yaml["spec"]["serviceAccountName"] = KUBERNETES_SERVICE_ACCOUNT
-            # use internal kfp static method to write the modified yaml back to the
-            # pipeline_file_path so we do not have to recreate kfp support for the
-            # various extensions it supports
-            kfp.compiler.Compiler()._write_workflow(workflow_yaml, pipeline_file_path)
+            self.update_kfp_yaml(
+                pipeline_file_path,
+                [
+                    ("spec.serviceAccountName", KUBERNETES_SERVICE_ACCOUNT),
+                ],
+            )
 
         return os.path.abspath(pipeline_file_path)
 
+    def update_kfp_yaml(
```

Comment:
please see above comment about how we can modify the workflow yaml dict in memory.
That would then no longer need this function because the modifications could then be simple dict updates as was done previously in:

```python
workflow_yaml["spec"]["serviceAccountName"] = KUBERNETES_SERVICE_ACCOUNT
```